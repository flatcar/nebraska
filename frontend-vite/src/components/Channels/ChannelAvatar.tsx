import { Theme } from '@mui/material';
import Avatar from '@mui/material/Avatar';
import makeStyles from '@mui/styles/makeStyles';

export interface ChannelAvatarProps {
  backgroundColor?: string;
  color?: string;
  size?: string | number;
  children?: React.ReactNode;
}

const useStyles = makeStyles((theme: Theme) => ({
  colorAvatar: (props: ChannelAvatarProps) => ({
    color: 'rgb(15 15 15)',
    backgroundColor: props.backgroundColor || props.color || theme.palette.secondary.main,
    width: props.size,
    height: props.size,
    display: 'inline-flex',
  }),
}));

export default function ChannelAvatar(props: ChannelAvatarProps) {
  const classes = useStyles(props);

  return <Avatar className={classes.colorAvatar}>{props.children || ' '}</Avatar>;
}
