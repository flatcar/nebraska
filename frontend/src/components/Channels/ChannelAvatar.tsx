import Avatar from '@mui/material/Avatar';

export interface ChannelAvatarProps {
  backgroundColor?: string;
  color?: string;
  size?: string | number;
  children?: React.ReactNode;
}

export default function ChannelAvatar({
  backgroundColor,
  color,
  size,
  children,
}: ChannelAvatarProps) {
  return (
    <Avatar
      sx={{
        color: 'rgb(15 15 15)',
        display: 'inline-flex',
        backgroundColor: theme => backgroundColor ?? color ?? theme.palette.secondary.main,
        width: size,
        height: size,
      }}
    >
      {children || ' '}
    </Avatar>
  );
}
