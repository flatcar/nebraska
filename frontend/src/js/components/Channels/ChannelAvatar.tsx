import { Theme } from '@material-ui/core';
import Avatar from '@material-ui/core/Avatar';
import makeStyles from '@material-ui/styles/makeStyles';
import React from 'react';

interface ChannelAvatarProps {
  backgroundColor?: string;
  color?: string;
  size?: number;
  children?: React.ReactNode;
}

const useStyles = makeStyles( (theme: Theme) => ({
  colorAvatar: (props: ChannelAvatarProps) => ({
    color: '#fff',
    backgroundColor: props.backgroundColor || props.color ||
    theme.palette.secondary.main,
    width: props.size,
    height: props.size,
    display: 'inline-flex'
  }),
}));

export default function ChannelAvatar(props: ChannelAvatarProps) {
  const classes = useStyles(props);

  return (
    <Avatar className={classes.colorAvatar}>{props.children || '' }</Avatar>
  );
}
