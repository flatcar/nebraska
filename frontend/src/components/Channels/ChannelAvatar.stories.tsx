import { Meta } from '@storybook/react';

import ChannelAvatar from './ChannelAvatar';

export default {
  title: 'channels/ChannelAvatar',
} as Meta;

export const White = () => <ChannelAvatar color="#fff">ABC</ChannelAvatar>;
