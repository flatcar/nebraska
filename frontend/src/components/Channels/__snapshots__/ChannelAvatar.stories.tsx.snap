// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channels/ChannelAvatar White smoke-test 1`] = `
<div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault stable-makeStyles-colorAvatar-1 stable-makeStyles-colorAvatar-2 css-jmdkhx-MuiAvatar-root">
  ABC
</div>
`;
